src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3290,6 +3289,21 @@ private bool HasTopLevelNullabilityConversion(TypeSymbolWithAnnotations source,
                     goto case ConversionKind.Identity;
 
                 case ConversionKind.Identity:
+                    // If the operand is an explicit conversion, and this identity conversion
+                    // is converting to the same type including nullability, skip the conversion
+                    // to avoid reporting redundant warnings. Also check useLegacyWarnings
+                    // since that value was used when reporting warnings for the explicit cast.
+                    if (useLegacyWarnings && operandOpt?.Kind == BoundKind.Conversion)
+                    {
+                        var operandConversion = (BoundConversion)operandOpt;
+                        var explicitType = operandConversion.ConversionGroupOpt.ExplicitType;
+                        if (!explicitType.IsNull && explicitType.Equals(targetTypeWithNullability, TypeCompareKind.ConsiderEverything))
```

Comment:
>ConsiderEverything [](start = 115, length = 18)

Should insignificant annotation differences be ignored for the purpose of this comparison?

Reply:
Perhaps, although it's not clear how to get `NullableBasedOnAnalysis` or `NotNullableBasedOnAnalysis` in either the target type or the explicit cast type. I'll leave as is, and we can adjust if there are insignificant differences in the future.